standalone/api.c

```diff
@@ -782,3 +786,61 @@ void modsecReportRemoteLoadedRules()
 
 }
 #endif
+
+#ifdef WAF_JSON_LOGGING_ENABLE
+void modsecReopenLogfileIfNeeded(request_rec *r)
+{
+    modsec_rec *msr = NULL;
+    int rc = 0;
+    apr_file_t * fd = NULL;
+    /* Find the transaction context first. */
+    msr = retrieve_msr(r);
+
+    if (msr == NULL)
+        return;
+
+    if (msc_waf_log_reopened){
+        if (msr->modsecurity != NULL && msr->modsecurity->wafjsonlog_lock != NULL){
+            rc = waf_get_exclusive_lock(msr->modsecurity->wafjsonlog_lock);
+            if (waf_lock_is_error(rc)) {
+                ap_log_rerror(APLOG_MARK, APLOG_ERR | APLOG_NOERRNO, 0, r,
+                        "ModSecurity not able to get lock for %s file", msc_waf_log_path);
+                return;
+            }
+
+            rc = apr_file_open(&fd, msc_waf_log_path,
+                   APR_WRITE | APR_APPEND | APR_CREATE | APR_BINARY,
+                   CREATEMODE | APR_WREAD, msc_waf_log_cmd->pool);
+
+            if (rc != APR_SUCCESS) {
+                ap_log_error(APLOG_MARK, APLOG_NOTICE, 0, NULL, "ModSecurity: " \
+                    "not able to reopen file: %s",
+                    msc_waf_log_path);
+
+                waf_free_exclusive_lock(msr->modsecurity->wafjsonlog_lock);
```

Comment:
the other places we are checking the return code of this call. Here we are not. Why?